test/integration/controllermanager/shoot/conditions/conditions_test.go

```diff
@@ -0,0 +1,191 @@
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Shoot Conditions controller tests", func() {
+	var (
+		shoot       *gardencorev1beta1.Shoot
+		managedSeed *seedmanagementv1alpha1.ManagedSeed
+		seed        *gardencorev1beta1.Seed
+	)
+
+	BeforeEach(func() {
+		shoot = &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-",
+				Namespace:    testNamespace.Name,
+				Labels:       map[string]string{testID: testRunID},
+			},
+			Spec: gardencorev1beta1.ShootSpec{
+				SecretBindingName: "my-provider-account",
+				CloudProfileName:  "cloudprofile1",
+				Region:            "europe-central-1",
+				Provider: gardencorev1beta1.Provider{
+					Type: "foo-provider",
+					Workers: []gardencorev1beta1.Worker{
+						{
+							Name:    "cpu-worker",
+							Minimum: 3,
+							Maximum: 3,
+							Machine: gardencorev1beta1.Machine{
+								Type: "large",
+							},
+						},
+					},
+				},
+				DNS: &gardencorev1beta1.DNS{
+					Domain: pointer.String("some-domain.example.com"),
+				},
+				Kubernetes: gardencorev1beta1.Kubernetes{
+					Version: "1.20.1",
+				},
+				Networking: gardencorev1beta1.Networking{
+					Type: "foo-networking",
+				},
+			},
+		}
+
+		By("Create Shoot")
+		Expect(testClient.Create(ctx, shoot)).To(Succeed())
+		log.Info("Created shoot for test", "shoot", client.ObjectKeyFromObject(shoot))
+
+		DeferCleanup(func() {
+			By("Delete Shoot")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, shoot))).To(Succeed())
+		})
+
+		managedSeed = &seedmanagementv1alpha1.ManagedSeed{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      shoot.Name,
+				Namespace: testNamespace.Name,
+				Labels:    map[string]string{testID: testRunID},
+			},
+			Spec: seedmanagementv1alpha1.ManagedSeedSpec{
+				Shoot: &seedmanagementv1alpha1.Shoot{
+					Name: shoot.Name,
+				},
+				Gardenlet: &seedmanagementv1alpha1.Gardenlet{},
+			},
+		}
+
+		seed = &gardencorev1beta1.Seed{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   managedSeed.Name,
+				Labels: map[string]string{testID: testRunID},
+			},
+			Spec: gardencorev1beta1.SeedSpec{
+				Provider: gardencorev1beta1.SeedProvider{
+					Region: "region",
+					Type:   "providerType",
+				},
+				Networks: gardencorev1beta1.SeedNetworks{
+					Pods:     "10.0.0.0/16",
+					Services: "10.1.0.0/16",
+					Nodes:    pointer.String("10.2.0.0/16"),
+					ShootDefaults: &gardencorev1beta1.ShootNetworks{
+						Pods:     pointer.String("100.128.0.0/11"),
+						Services: pointer.String("100.72.0.0/13"),
+					},
+				},
+				DNS: gardencorev1beta1.SeedDNS{
+					IngressDomain: pointer.String("someingress.example.com"),
+				},
+			},
+		}
+	})
+
+	Context("preconditions not fulfilled", func() {
+		AfterEach(func() {
```

Comment:
I would propose to rather use some shared assertion function instead of doing the "main" assertions in `AfterEach`.
Executing main assertions in `AfterEach` or `BeforeEach` messes up the machine-readable reports, because failures are not reported on the `It` itself.